src/app/app.component.ts

```diff
@@ -10,25 +16,95 @@ import { AuthService } from './auth/auth.service';
 export class AppComponent {
   title = 'altrusight';
   sidebarOpened: boolean;
+  allNonProfitsNames: string[] = [];
+
+  // searchInput = new FormControl();
+  // filteredNonProfits: Observable<string[]>;
+
 
   // TODO: issue with auth --> auth service is not instantiated until AFTER app component is
   // so we will always have auth issues on first render until fixing this
   // https://stackoverflow.com/questions/35191617/how-to-run-a-service-when-the-app-starts-in-angular-2
-  constructor(public router: Router, public authService: AuthService) {
+  constructor(public router: Router, public authService: AuthService, private nonProfitsService: NonprofitsService) {
       this.sidebarOpened = this.authService.isLoggedIn;
+      this.nonProfitsService.getNonprofit('')
+
+      // KEEP CODE BELOW: In case we want to improve search functinality
+      // as a drop down -->
+      // Initialize array that will hold nonprofit names
+      // this.initializeNonProfitsNamesArr();
+      // this.filteredNonProfits = this.searchInput.valueChanges.pipe(
+      //   // startWith(''),
+      //   map(value => this._filter(value))
+      // );
+  }
+
+  //========================================================
+  // Helper function to filer nonprofits as user types. 
+  // This function wil trigger on user input
+  //========================================================
+  private _filter(searchParam: string): string[] {
+    if(searchParam.length != 0) {
+
+      const filterValue = searchParam.toLowerCase();
+      return this.allNonProfitsNames.filter(option => option.toLowerCase().indexOf(filterValue) >= 0);    
+    }
+
+    return [];
+  }
+
+  //========================================================
+  // Will populate nonprofits names array with results of API
+  // call.
+  // TODO: API call returns only 100 nonprofits (in alphabetical order).
```

Comment:
I see you noted this. Yeah I'll get together with you and whoever else decides to look at the API and we can brainstorm how to get past this limit. The actual API has a limitation of 1K and they don't have to be alphabetical, but in the backend for testing I have it set to 100 alphabetically. Either way the solution we come up with will address both problems. This will be a priority for me and I'll try to get that fixed by next week.